build/js/inverted-index.js

```diff
@@ -14,154 +14,168 @@ class InvertedIndex {
 
 
   /**
-   * converts the string provided to lower case, strips the string of all 
+   * converts the string provided to lower case, strips the string of all
    * special characters, extra spaces and converts the result into an Array
-   * @static 
    * @param {string} text the text to be converted
    * @returns {Array} the result that will be returned after function call
    */
-  static textToArray(text) {
-    const result = text.toLowerCase().replace(/[^a-z0-9\s]/g, "").split(/\s+/);
+  textToArray(text) {
+    const result = text.toLowerCase().replace(/[^a-z0-9\s]/g, '').split(/\s+/);
     return result;
   }
 
-
   /**
-   * loops through an array and removes duplicate elements
-   * @static
-   * @param {}
-   * @returns {Array} 
+   * check the documents in the uploaded file and returns
+   * the content of the file in a tokenized format
+   * @param {Array} file the content of the file
+   * @returns {Array} returns a boolean
    */
-  _sortIndex(bookName, arrayItem, element) {
-    element = parseInt(element);
-    for(let item in arrayItem) {
-      if(!this.indexes[bookName].hasOwnProperty([arrayItem[item]])) {
-        this.indexes[bookName][arrayItem[item]] = [];
-        this.indexes[bookName][arrayItem[item]].push(element); 
-      } else if(
-        this.indexes[bookName].hasOwnProperty([arrayItem[item]]) && 
-        this.indexes[bookName][arrayItem[item]].indexOf(element) === -1) {
-        this.indexes[bookName][arrayItem[item]].push(element);
+  tokenizeFile(file) {
+    const fileContent = [];
+    for (let i = 0; i < file.length; i += 1) {
+      if (file[i].title && file[i].text) {
+        const text = `${file[i].title} ${file[i].text}`;
+        const fileDoc = this.textToArray(text);
+        fileContent.push(fileDoc);
       }
     }
+    return fileContent;
   }
 
   /**
-   * reads the content of a json file and returns false if the file is empty or 
-   * returns true if there are documents in the file.
-   * @param {Array} book the content of the book
+   * checks the content of the uploaded json file and returns
+   * true if the file follows the allowed format
+   * @param {Array} file the content of the file
    * @returns {boolean} returns a boolean
-   * 
    */
-  readBookData(book) {
-    if(book.length < 1) {
+  checkBookData(file) {
+    if (!Array.isArray(file) || file.length < 1) {
       return false;
-    } else {
-      return true;
+    }
+    for (let i = 0; i < file.length; i += 1) {
```

Comment:
the ESLint rule for the checkpoint flags an error if the incrementor is used that way